components/nexusoperations/executors.go

```diff
@@ -353,6 +366,117 @@ func (e activeExecutor) executeTimeoutTask(ctx context.Context, env hsm.Environm
 	})
 }
 
+func (e activeExecutor) executeCancelationTask(ctx context.Context, env hsm.Environment, ref hsm.Ref, task CancelationTask) error {
+	service, err := e.NamespaceRegistry.NexusOutgoingService(namespace.ID(ref.WorkflowKey.NamespaceID), task.Destination)
+	if err != nil {
+		if errors.As(err, new(*serviceerror.NotFound)) {
+			// The mapping doesn't exist. Assume the registry's NexusOutgoingService will readthrough to verify this
```

Comment:
Since outgoing services are part of the namespace entry, why do we treat service not found and namespace not found errors differently?

Reply:
If the namespace is not found, there's not much we can do. We can't really process workflows on that namespace anymore. I _think_ it's okay to drop tasks - which is the current behavior.

If a service is deleted we can decide what to do but I figured it's better to fail immediately than to retry these tasks indefinitely. Dropping the task is obviously not the right behavior.